src/shaders/_prelude.vertex.glsl

```diff
@@ -16,12 +16,37 @@ precision highp float;
 
 #endif
 
+#ifndef EXTENT
+#define EXTENT 8192.0
+#endif
+
+#ifndef PI
+#define PI 3.141592653589793
+#endif
+
 float wrap(float n, float min, float max) {
     float d = max - min;
     float w = mod(mod(n - min, d) + d, d) + min;
     return (w == min) ? max : w;
 }
 
+vec3 mix_globe_mercator(mat4 matrix, vec2 tile_anchor, vec3 position, vec3 tile_id, vec2 mercator_center, float t) {
+#ifdef PROJECTION_GLOBE_VIEW
+    float tiles = pow(2.0, tile_id.z);
+
+    vec2 mercator = (tile_anchor / EXTENT + tile_id.xy) / tiles;
+    mercator -= mercator_center;
+    mercator.x = wrap(mercator.x, -0.5, 0.5);
+
+    vec4 mercator_tile = vec4(mercator.xy * EXTENT, EXTENT / (2.0 * PI), 1.0);
+    mercator_tile = matrix * mercator_tile;
+
+    return mix(position, mercator_tile.xyz, vec3(t));
+#else
+    return position;
+#endif
+}
```

Comment:
So if I understood right you're using the inversion matrix for taking the mercator location of the coordinate in to the encoded space? And this way you can use the same globe matrix for both points?

Reply:
Yes exactly, this is the simplest approach that I have found. To be able to do that using the same globe projection matrix later on we need to revert the rotation + encoding so that we can put these vertices in a flat plane using that same matrix. The other approach would be to transform all of these points in that space on the CPU _and_ pass another projection per tile coordinate, but that ends being a lot of extra changes for the same result.